tutorials/plot_stats.py

```diff
@@ -11,40 +11,126 @@
 # Applying statistical measures to spiking data
 # ---------------------------------------------
 #
-# Words, words, words.
+# Sections
+# ~~~~~~~~
+#
+# This tutorial contains the following sections:
+#
+# 1. Compute and plot different shuffles of spikes
+# 2. Compute empirical p-value from distribution of spikes
+# 3. Compute z-score from distribution of surrogates
 #
 
 ###################################################################################################
 
 # sphinx_gallery_thumbnail_number = 1
 
-import spiketools
+# import auxiliary libraries
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+# import all functions from spiketools.stats
+from spiketools.stats.generators import *
+from spiketools.stats.permutations import *
+from spiketools.stats.shuffle import *
 
 ###################################################################################################
 
-# Settings
-...
+# Generate spike times for spikes at 10Hz for 1000 seconds
+poisson_generator = poisson_train(10, 1000)
+# get spike in seconds
+spikes_s = np.array([spike for spike in poisson_generator])
+# convert to milliseconds
+spikes_ms = spikes_s*1000
+spikes_ms = np.unique(spikes_ms.astype(int))
 
 ###################################################################################################
-# Subtitle
+# 1. Compute and plot different shuffles of spikes
 # ~~~~~~~~
 #
-# Words, words, words.
+# Use spikes_ms to compute shuffle_isis (with 10 shuffles), shuffle_bins (with 10 shuffles), 
+# shuffle_poisson (with 10 shuffles), and shuffle_circular (with 10 shuffles).
+# Also compute a vector permutation, for comparison.
 #
 
 ###################################################################################################
 
-# Do some code stuff
-print('Hello World!')
+# Shuffle spike ms using four methods
+shuffled_isis = shuffle_isis(spikes_ms, n_shuffles=10)
+shuffled_bins = shuffle_bins(spikes_ms, bin_width_range=[50, 200], n_shuffles=10)
+shuffled_poisson = shuffle_poisson(spikes_ms, n_shuffles=10)
+shuffled_circular = shuffle_circular(spikes_ms, shuffle_min=200, n_shuffles=10)
+
+# Do a vector permutation, for comparison
+perm_spikes = vec_perm(spikes_ms, n_perms=10)
 
 ###################################################################################################
 
-# Do more code stuff
-print('More stuff!')
+# Plot non-shuffled, all different shuffles, and vector permutation
+f, ((ax1, ax2, ax3), (ax4, ax5, ax6)) = plt.subplots(2, 3, sharey = True)
+# original
+ax1.eventplot(spikes_ms[0:50])
+ax1.set_title('Non-shuffled')
+ax1.set_xlim([0, 6000])
+ax1.set_xlabel('time')
+
+# isis
+ax2.eventplot(shuffled_isis[:, 0:80])
```

Comment:
In general, I think we can use the plot function `plot_trial_rasters` available in `spiketools`, and it should save us a bunch of code. 

Something like this should work, I think:
`plot_trial_rasters(spikes_ms[0:50], xlim=[0, 6000], ax=ax2)